### ISSUE TITLE
🚀 Enhancement: Add Hyperparameter Tuning Section

### ISSUE THREAD

**Issue by rohan-vivek** [1998-aish/churn-prediction-ml#4]
## 💡 Suggestion: Improve Model Performance

Your current models use good default parameters, but hyperparameter tuning could push performance even higher!

### 📊 Current Best Model
- **Logistic Regression**: 80.6% accuracy, 0.56 recall

### 🎯 Proposed Enhancement

Add GridSearchCV or RandomizedSearchCV to find optimal hyperparameters:

```python
from sklearn.model_selection import GridSearchCV

# Example for Random Forest
param_grid = {
    'classifier__n_estimators': [100, 200, 300],
    'classifier__max_depth': [None, 10, 20, 30],
    'classifier__min_samples_split': [2, 5, 10],
    'classifier__class_weight': ['balanced', 'balanced_subsample']
}

grid_search = GridSearchCV(
    rf_model,
    param_grid,
    cv=5,
    scoring='recall',  # Optimize for churn recall
    n_jobs=-1
)

grid_search.fit(X_train, y_train)
print(f"Best parameters: {grid_search.best_params_}")
print(f"Best recall score: {grid_search.best_score_}")
```

### 🎁 Expected Benefits
- 📈 Potentially improve recall from 0.56 to 0.60+
- 🎯 Find optimal model configuration
- 📝 Document best practices

### ⚠️ Note
This will increase training time, so you might want to use `RandomizedSearchCV` for faster results with XGBoost.

Would this be useful? Happy to contribute! 🙌
